[신세원 Feature/detail_item 컴포넌트 레이아웃 구현완료

shinsewon opened the pull request:
## 수정 사항 간략한 한줄 요약
- Feature/detail_item 레이아웃 구현완료
- 판단 하에 컴포넌트화 가능 구역 분리작업 완료
- 각 컴포넌트 별 반복 구간 map 메소드 사용 완료
		
## 수정 사항들 자세한 내용
- 상기와 동일

## 기타 질문 및 특이 사항
- 아직까진 없습니다.

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

shinsewon committed:
[detail_item]
1.레이아웃 구현중

shinsewon committed:
[detail_item]
1. 아이템 상세항목 레이아웃 구현중

shinsewon committed:
수정

shinsewon committed:
[신세원]
1.상품 디테일 페이지 레이아웃 구현중
2.일부 구역들 컴포넌트화 작업 진행중

shinsewon committed:
브랜치 변경 위한 임시 커밋 및 select ui 설치

shinsewon committed:
[신세원]
1.상품 디테일 페이지 레이아웃 구현완료
2.상품 디테일 페이지 상세 컴포넌트화 완료
3.각 컴포넌트별 반복 구간 map 메소드 사용완료

shinsewon committed:
[신세원]
1.상품 디테일 페이지 기능 구현중

bonbon0202 commented on src/Components/DetailItem/DetailItem.js:
```diff
@@ -0,0 +1,25 @@
+import './DetailItem.scss';
+
+class DetailItem extends Component {
+  constructor(props) {
+    super(props);
+  }
```
필요한 코드일까요?! 🤔 불필요하다면 삭제!

shinsewon replied:
삭제했습니다!

bonbon0202 commented on src/Components/DetailItem/DetailItem.scss:
```diff
@@ -0,0 +1,18 @@
+@mixin flexBox {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
```
굿굿!! ㅎㅎ 잘 활용해주셨네요! 
위 내용을 common.scss에 적어주시면 모든 파일에서 이용가능하답니다 :)

shinsewon replied:
아! 알겠습니다!! 적용하겠습니다!

bonbon0202 commented on src/Components/DetailItem/DetailItem.scss:
```diff
@@ -0,0 +1,18 @@
+    @include flexBox;
+    width: 1300px;
+    padding-top: 70px;
+    margin: 0 310px 0 310px;
```
```suggestion
    margin: 0 310px;
```

shinsewon replied:
수정했습니다!

bonbon0202 commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,153 @@
+import React, { Component } from 'react';
+import Select from 'react-select';
+import './DropDownBox.scss';
+import OrderBox from './OrderBox';
```
import 에도 가독성을 위해 지켜야할 컨벤션 순서가 있답니다! 
패키지 > 컴포넌트 > 이미지 > css 순서로 import 작성해주세요 :)

shinsewon replied:
네 알겠습니다!

bonbon0202 commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,153 @@
+
+  //사이즈 클릭시 값이 누적되고 오더박스 카드가 생김
+  handleSizeCheck = (e) => {
+    // console.log('e는 뭐지?', e);
```
'이건 뭐지?'
ㅎㅎㅎ 콘솔 찍어보시는거 굿굿! 그러나 사용 후엔 삭제해주세요!

shinsewon replied:
네 알겠습니다!

bonbon0202 commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,153 @@
+  }
+
+  componentDidMount() {
+    this.setState({});
```
??

shinsewon replied:
지웠습니다!

bonbon0202 commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,153 @@
+    let arr = [...item];
+    arr.push(selectPrice);
+    // console.log('arr에 item[] 복사됨 : ', arr);
+    let totalPrice = arr.reduce((acc, obj) => acc + obj, 0);
```
reduce 활용 굿굿!! ☺️ 아래와 같이 사용할 수 있겠네요. 
```suggestion
    let totalPrice = arr.reduce((acc, obj) => acc + obj);
```

shinsewon replied:
지금도 코드가 바껴서 저 코드랑은 달라졌습니다 나중에 응용해보겠습니다 :)

bonbon0202 commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,153 @@
+    // console.log('arr에 item[] 복사됨 : ', arr);
+    let totalPrice = arr.reduce((acc, obj) => acc + obj, 0);
+    // console.log('totalPrice에 값이 더해짐 :', totalPrice);
+    let orderBox2 = [...orderBox];
+    orderBox2.push({ value, size: e.value, priceBox: 16200 });
```
아래와 같이 적어주시면 한번에 적어주실 수 있습니다. 
```suggestion
      let orderBox2 = [...orderBox, { value, size: e.value, priceBox: 16200 }];
```

shinsewon replied:
저렇게 적용할 수가 있네요!
지금은 코드가 많이 바뀌어서 다르게 적용했습니다.
다음엔 저렇게 적용해보겠습니다!

bonbon0202 commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,153 @@
+    let orderBox2 = [...orderBox];
+    orderBox2.push({ value, size: e.value, priceBox: 16200 });
+    //중복된 배열이 있을때 걸러주는 기능 추가
+    const filterItem = orderBox2;
+    const result = (a, b) => {
+      const newArray = [];
+      const obj = {};
+      for (let i in a) {
+        obj[a[i][b]] = a[i];
+      }
+      for (let i in obj) {
+        newArray.push(obj[i]);
+      }
+      return newArray;
+    };
+    const uniqueArray = result(filterItem, 'size');
+
+    this.setState({
+      value,
+      size: e.label,
+      click2: !click2,
+      defaultPrice: selectPrice,
+      item: arr,
+      priceBox: totalPrice,
+      orderBox: uniqueArray,
+    });
```
그런데 합계 값은 select를 누를 때마다 계속 누적이되고 있네요..!

shinsewon replied:
수정했습니다 :)

bonbon0202 commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,153 @@
+
+  render() {
+    const { click, value, size, priceBox, click2, number } = this.state;
+    console.log('orderBox :', this.state.orderBox);
```
확인한 콘솔은 삭제!

shinsewon replied:
삭제완료!

bonbon0202 commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,153 @@
+  render() {
+    const { click, value, size, priceBox, click2, number } = this.state;
+    console.log('orderBox :', this.state.orderBox);
+
```
빈 줄 삭제!

shinsewon replied:
삭제했습니다!

bonbon0202 commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,153 @@
+          {this.state.orderBox.map((el) => {
+            return <OrderBox value={el.value} size={el.size} priceBox={el.priceBox} />;
+          })}
+          {/* {click2 && <OrderBox value={value} size={size} priceBox={priceBox} />} */}
+
```
삭제!

shinsewon replied:
넵!!

bonbon0202 commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,153 @@
+          )}
+        </form>
+        <footer className="footer">
+          {this.state.orderBox.map((el) => {
```
map 메소드 잘 활용해주셨는데 구조분해할당도 잘 해주세요 ><

shinsewon replied:
수정했습니다!

bonbon0202 commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,153 @@
+            className="select1"
+            onChange={this.handleChange}
+            options={COLLOR}
+            defaultValue={COLLOR[0]}
```
굿굿!

shinsewon replied:
감사합니다!

bonbon0202 commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,153 @@
+            options={COLLOR}
+            defaultValue={COLLOR[0]}
+          />
+          {number > 0 && (
+            <Select
+              className="select2"
+              onChange={this.handleSizeCheck}
+              options={SIZE}
+              defaultValue={SIZE[0]}
+            />
+          )}
```
boolean값을 이용해 아래와 같이 작성해주실 수 있겠네요! 
```suggestion
          {number && (
            <Select
              className="select2"
              onChange={this.handleSizeCheck}
              options={SIZE}
              defaultValue={SIZE[0]}
            />
          )}
```

shinsewon replied:
저렇게 쓰면 아래에 '0'이 그대로 출력됩니다ㅜㅜ

bonbon0202 commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,153 @@
+            <div className="directPurchase">바로 구매</div>
+            <div className="logoContainer">
+              <div className="logoBox">
+                <img className="naverLogo" src="./images/naver.png" alt="네이버 로고" />
```
nesting 속성을 이용한다면 굳이 className이 필요할것 같지 않네요! 
```css
     .logoBox{
          img{}
     }
```

shinsewon replied:
바꿨습니다!

bonbon0202 commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,153 @@
+
+export default DropDownBox;
+
+const COLLOR = [
+  { value: 'default', label: '[컬러]를 선택하세요.' },
+  { value: 'black', label: '블랙' },
+  { value: 'yellow', label: '옐로우' },
+  { value: 'gray', label: '그레이' },
+];
+
+const SIZE = [
+  { value: 'default', label: '[사이즈]를 선택하세요.' },
+  { value: 'small', label: 'Small' },
+  { value: 'medium', label: 'Medium' },
+  { value: 'large', label: 'Large' },
+];
+
+const TEST = [
+  '블랙small',
+  '블랙medium',
+  '블랙large',
+  '옐로우small',
+  '옐로우medium',
+  '옐로우large',
+  '그레이small',
+  '그레이medium',
+  '그레이large',
+];
```
data.js에 넣어서 관리해주세요!

shinsewon replied:
STAR.js 파일 만들어서 관리하였습니다!

shinsewon replied:
ITEM.js 만들어서 관리하였습니다!

bonbon0202 commented on src/Components/DetailItem/DropDownBox.scss:
```diff
@@ -0,0 +1,109 @@
+      padding: 0 30px;
+      width: 100%;
+      height: 44px;
+      // border: 1px solid red;
```
삭제!

shinsewon replied:
삭제완료!

bonbon0202 commented on src/Components/DetailItem/ItemSection.js:
```diff
@@ -0,0 +1,72 @@
+import './ItemSection.scss';
+
+class ItemSection extends Component {
+  constructor(props) {
+    super(props);
+  }
```
삭제!

shinsewon replied:
삭제했습니다!

bonbon0202 commented on src/Components/DetailItem/ItemSection.js:
```diff
@@ -0,0 +1,72 @@
+
+export default ItemSection;
+
+const STAR = [
+  { img: './images/star.png' },
+  { img: './images/star.png' },
+  { img: './images/star.png' },
+  { img: './images/star.png' },
+  { img: './images/star.png' },
+];
```
data.js에 따로 파일을 분리해서 관리해주세요! :)

shinsewon replied:
STAR.js 파일 만들어서 관리하였습니다!

bonbon0202 commented on src/Components/DetailItem/ItemSection.scss:
```diff
@@ -0,0 +1,138 @@
+.section {
+  width: 100%;
+
```
??

shinsewon replied:
100%로 안하면 화면의 비율이 달라집니다 ㅜㅜ 밑에 다른 요소들 추가 했습니다!

bonbon0202 commented on src/Components/DetailItem/ItemSection.scss:
```diff
@@ -0,0 +1,138 @@
+  .header {
+    display: flex;
+    flex-direction: column;
+    .storeBox {
+      display: flex;
+      margin-bottom: 10px;
+      .store {
```
선택자 간에는 한 줄 띄어주세요!

shinsewon replied:
수정했습니다!

bonbon0202 commented on src/Components/DetailItem/OrderBox.js:
```diff
@@ -0,0 +1,43 @@
+            </div>
+            <div className="orderBottomBox">
+              <div className="orderBottom">
+                <button className="minus">-</button>
+                <div className="number">1</div>
+                <button className="plus">+</button>
```
기능구현은 앞으로 추가 하시는거죠?ㅎㅎ

shinsewon replied:
네네 했습니다 :)

bonbon0202 commented on src/Components/DetailItem/OrderBox.js:
```diff
@@ -0,0 +1,43 @@
+                <div className="delivery">일반배송</div>
+              </div>
+              <div className="buttonBox">
+                <button>X</button>
```
영문 x보다는 아이콘을 사용해보시는 건 어떨까요..?ㅎㅎㅎㅎ

shinsewon replied:
아이콘으로 바꿨습니다!

bonbon0202 commented on src/Components/DetailItem/OrderBox.js:
```diff
@@ -0,0 +1,43 @@
+class OrderBox extends Component {
+  render() {
+    const { value, size, priceBox } = this.props;
+    // console.log('props.value', value);
+    // console.log('props.size', size);
+    // console.log('props.priceBox', priceBox);
+    // console.log('========================');
```
콘솔은 어떻게??ㅋㅋ

shinsewon replied:
불필요한 콘솔 지웠습니다 ㅎㅎ

bonbon0202 reviewed (commented):
세원님 :) 수고 많으셨어요!
주석, 콘솔, 불필요한 빈 줄, css 선택자 간에는 한 줄 띄어주기, data,js 파일분리! 
ㅎㅎㅎㅎ 꼭 지켜주세요...! push 전에 한번씩 확인하기!

shinsewon committed:
[신세원]
1.상품 디테일 페이지 상품 구매 박스 기능 구현완료

shinsewon committed:
디테일 홈페이지 상품 주문 수량 기능 구현중

shinsewon committed:
[신세원]
1.detail_item 브랜치 기능 구현중
2.상품 오더박스 주문 수량에 따른 누적 합산 금액 기능 구현완료
3.오더 박스 삭제 기능 구현완료
4.오더 박스 +.- 버튼 수량 조절 기능 구현완료
5.찜 버튼 클릭시 이미지 체인지 기능 구현완료

shinsewon committed:
Fix: conflict resolved

shinsewon committed:
브랜치 이동을 위한 임시 커밋

shinsewon committed:
코드리뷰 반영 완료

shinsewon committed:
Fix: conflict resolved

shinsewon committed:
필요없는 컴포넌트 제거

shinsewon committed:
새로운 브랜치 생성을 위한 커밋

shinsewon committed:
[detail_item]
1.코드리뷰 반영 및 수정완료

dipsiiiiiiiiii commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,174 @@
+    };
+  }
+
+  componentDidUpdate(prevprops, prevstate) {
+    const { orderBox } = this.state;
+    if (prevstate.orderBox !== this.state.orderBox) {
+      const priceBox = orderBox.reduce((a, b) => {
+        a = a + b.priceBox * b.amount;
+        return a;
+      }, 0);
+      this.setState({ priceBox });
+    }
+  }
```
👏👏👏
너무 잘 하셨는데 reduce 쓰실 때 a, b로 인자 하지 마시고 의미 있는 변수 이름으로 해주세요!

shinsewon replied:
acc,prev 값으로 변경했습니다!

dipsiiiiiiiiii commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,174 @@
+    arr.push(selectPrice);
+    //중복된 배열이 있을때 걸러주는 기능 추가
+    let orderBox2 = [...orderBox];
+    let orderBox3 = orderBox2.filter((el) => (el.value === value && el.size === e.label ? el : ''));
```
filter는 return값의 Boolean을 비교해서 true면 해당 요소를 그대로 반환합니다 그래서 아래와 같이 적어주시면 됩니다.
```suggestion
    let orderBox3 = orderBox2.filter((el) => (el.value === value && el.size === e.label));
```

shinsewon replied:
좋은정보 감사합니다 바로 반영하였습니다!

dipsiiiiiiiiii commented on src/Components/DetailItem/ItemSection.js:
```diff
@@ -0,0 +1,73 @@
+
+export default ItemSection;
+
+// const STAR = [
+//   { id: 1, img: './images/star.png' },
+//   { id: 2, img: './images/star.png' },
+//   { id: 3, img: './images/star.png' },
+//   { id: 4, img: './images/star.png' },
+//   { id: 5, img: './images/star.png' },
+// ];
```
주석 제거해주세요!

shinsewon replied:
제거하였습니다!

dipsiiiiiiiiii commented on src/Components/DetailItem/DropDownBox.js:
```diff
@@ -0,0 +1,174 @@
+    this.setState({ orderBox: orderBoxMap });
+  };
+
+  handleMinus = (el) => {
+    const { orderBox } = this.state;
+    let orderBoxMinus = orderBox.map((orderbox) => {
+      if (el.idx === orderbox.idx) {
+        orderbox.amount = orderbox.amount - 1;
+      }
+      return orderbox;
+    });
+    if (orderBoxMinus[orderBoxMinus.indexOf(el)].amount < 1) {
+      return (orderBoxMinus[orderBoxMinus.indexOf(el)].amount = 1);
+    }
+    this.setState({ orderBox: orderBoxMinus });
+  };
```
handlePlus와 handleMinus에서 공통된 로직을 합쳐볼 수 있을 것 같습니다!

shinsewon replied:
하나는 더해야하는 이벤트이고, 하나는 빼야하는 독립적인 이벤트인데 어떻게 합칠수 있을까요?ㅜㅜ
둘다 로직은 같지만 리턴 값이 달라서 어떻게 합쳐야 할지 모르겠습니다.
인터넷에도 독립적으로 이벤트를 주더라고요 ㅜㅜ

dipsiiiiiiiiii commented on src/Components/DetailItem/OrderBox.js:
```diff
@@ -0,0 +1,59 @@
+import './OrderBox.scss';
+
+class OrderBox extends Component {
+  constructor(props) {
+    super(props);
+  }
```
state 안 쓰시면 지워주세요!

shinsewon replied:
삭제했습니다!

dipsiiiiiiiiii commented on src/Components/DetailItem/OrderBox.scss:
```diff
@@ -0,0 +1,120 @@
+
+          .minus,
+          .plus {
+            display: flex;
+            align-items: center;
+            justify-content: center;
```
mixin 활용해보실 수 있겠네요!

shinsewon replied:
mixin 사용하여 반영했습니다!

dipsiiiiiiiiii commented on src/Components/DetailItem/PictureBox.js:
```diff
@@ -0,0 +1,41 @@
+import './PictureBox.scss';
+
+class PictureBox extends Component {
+  constructor(props) {
+    super(props);
+  }
```
여기도 안쓰시면 삭제!

shinsewon replied:
삭제했습니다!

dipsiiiiiiiiii commented on src/Components/DetailItem/PictureBox.js:
```diff
@@ -0,0 +1,41 @@
+            <div className="pictureListBox">
+              {PICTURE_LIST.map((picture) => {
+                return (
+                  <div className="pictureList">
```
key 넣어주세요!

shinsewon replied:
바로 반영했습니다!

dipsiiiiiiiiii reviewed (commented):
고생하셨습니다
리뷰 반영해주세요!

shinsewon committed:
Fix: conflict resolved

shinsewon committed:
Fix: conflict resolved

shinsewon committed:
Fix: conflict resolved
